test/seqan3-test.cmake

```diff
@@ -26,6 +26,20 @@ option (SEQAN3_TEST_BUILD_OFFLINE "Skip the update step of external projects." O
 # have to be adapted or the option deactivated.
 option (SEQAN3_BENCHMARK_ALIGN_LOOPS "Pass -falign-loops=32 to the benchmark builds." ON)
 
+# ----------------------------------------------------------------------------
+# Custom Build types
+# ----------------------------------------------------------------------------
+
+# -DCMAKE_BUILD_TYPE=FEDORA; our library did not compile for fedora quite a few times, because of that we created this
+# custom build type to emulate their flag set-up.
+# We omitted:
+#   -specs=/usr/lib/rpm/redhat/redhat-hardened-cc1
+#   -specs=/usr/lib/rpm/redhat/redhat-annobin-cc1
+#   set -mtune=native
+#   and -fcf-protection=check
```

Comment:
We could actually do a compile check for this one if `-fcf-protection=full` does not work, we use `=check`. 
But I guess this flag doesnt add much

Reply:
Yeah true, I think most of the flags are unneeded. I think we should have for a first version have only flags work on multiple platforms. Those could be refined later, if they show hidden errors.